### PR TITLE
Fix deserialization where the type hint from a previous object was being...

### PR DESCRIPTION
**Pull request by Prognathous**
... used for the next object that doesn't have a type hint.

### CODE REVIEW

**Comment by tritao** [comment]
Can you extend the test suite to cover this case?

File: mcs/class/System.ServiceModel.Web/System.Runtime.Serialization.Json/JsonReader.cs [hunk 1]
```diff
@@ -817,7 +817,7 @@ void ReadStartObject ()
 			if (PeekChar () == '"') { // it isn't premise: the object might be empty
 				ReadChar ();
 				string s = ReadStringLiteral ();
-				if (s == "__type") {
+                if (s == "__type") {
```
Comment (alexrp) [inline mcs/class/System.ServiceModel.Web/System.Runtime.Serialization.Json/JsonReader.cs]: unrelated style change

File: mcs/class/System.ServiceModel.Web/System.Runtime.Serialization.Json/JsonReader.cs [hunk 2]
```diff
@@ -826,8 +826,10 @@ void ReadStartObject ()
 					SkipWhitespaces ();
 					ei.HasContent = true;
 				}
-				else
-					next_object_content_name = s;
+                else {
```
Comment (alexrp) [inline mcs/class/System.ServiceModel.Web/System.Runtime.Serialization.Json/JsonReader.cs]: please use tabs instead of spaces for indentation

**Comment by esdrubal** [comment]
We already moved to use reference JSON source serializer, this happen on 54f7c82.
Modified files no longer exist since 8c634ba.

If this is still an issue please report it at https://bugzilla.xamarin.com/ with a use case.
